### PR TITLE
[New Rule] Conditional Returns on Newline

### PR DESCRIPTION
**Pull request by HaloZero**
Added an opt-in rule to enforce that guard's must return on new lines separate from guard statement. Opt-in as it's purely stylistic but enforced at my place of work and seems to also follow Github's style convention as well.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/GuardReturnsOnNewline.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+//  SwiftLint
+//
+//  Created by Rohan Dhaimade on 12/08/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct GuardReturnsOnNewline: Rule, OptInRule {
```
Comment (keith) [inline Source/SwiftLintFramework/Rules/GuardReturnsOnNewline.swift]: Maybe this rule should apply to if statements as well?

Comment (HaloZero) [inline Source/SwiftLintFramework/Rules/GuardReturnsOnNewline.swift]: Better name then: ConditionalReturnsOnNewLine?

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,9 @@
   [bootstraponline](https://github.com/bootstraponline)
   [#689](https://github.com/realm/SwiftLint/issues/689)
 
+* Adds 'GuardReturnsOnNewLineRule' rule.
```
Comment (jpsim) [inline CHANGELOG.md]: missing two trailing spaces as indicated by https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md#tracking-changes

File: Source/SwiftLintFramework/Models/MasterRuleList.swift [hunk 3]
```diff
@@ -43,6 +43,7 @@ public struct RuleList {
 public let masterRuleList = RuleList(rules:
     ClosingBraceRule.self,
     ColonRule.self,
+    ConditionalReturnsOnNewline.self,
```
Comment (jpsim) [inline Source/SwiftLintFramework/Models/MasterRuleList.swift]: not in alphabetical order

File: Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+//  SwiftLint
+//
+//  Created by Rohan Dhaimade on 12/08/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ConditionalReturnsOnNewline: Rule, OptInRule {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift]: Could you please make this conform to `ConfigurationProviderRule` with a `public var configuration = SeverityConfiguration(.Warning)` property? See [ForceUnwrappingRule.swift](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/ForceUnwrappingRule.swift) as an example.

You also don't need the `init(configuration:)` below.

**Comment by jpsim** [comment]
This looks good @HaloZero. Please let me know when you've had the chance to address my feedback and I'll take a final look.

File: Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift (hunk 4 above)
Comment (HaloZero) [inline Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift]: Sounds good. I'm planning on adding more stuff. Could I get some background on what this does and why it's necessary? I figured I didn't need it as I don't have any configurations.

**Comment by jpsim** [comment]
Wonderful, this looks ready to merge to me @HaloZero. Let me know when you're done making updates and I'll merge.

**Comment by HaloZero** [comment]
@jpsim I'm all done with any changes. Thanks!

**Comment by jpsim** [comment]
🎉 big thanks for your contribution

**Pull request merged**